src/doc/style-guide/src/expressions.md

```diff
@@ -512,10 +515,11 @@ if let ... {
 }
 ```
 
-If the control line needs to be broken, then prefer to break before the `=` in
-`* let` expressions and before `in` in a `for` expression; the following line
-should be block indented. If the control line is broken for any reason, then the
-opening brace should be on its own line and not indented. Examples:
+If the control line needs to be broken, then prefer breaking before the `=` for any
+`let` sub-expression in an `if` or `while` expression that does not fit,
+and before `in` in a `for` expression; the following line should be block indented.
+If the control line is broken for any reason, then the opening brace should be on its
+own line and not indented. Examples:
```

Comment:
This is one of the aspects that's been a struggle for me, as I feel like there's some important missing text here even without the additions for let-chains.

Specifically, what happens of the pattern itself ends up needing to be formatted across multiple lines?

The rules for let statements cover this scenario, and I imagine (though didn't check) rustfmt would do the same for a non-chained `if let`. However, the written rules here for expressions do already differ from those for statements (notably whether to break before or after the assignment operator) so I think this is an opportunity for increased clarity and explicitness we should consider for the 2024 edition, if not earlier.

I was tempted to create a new section that covered the `let` expressions explicitly, but ultimately decided it would be ideal if in the future we could consolidate the rules between the statement and expression contexts, so have punted for now